### ISSUE TITLE
FORD fails to read format string containing regular expression

### ISSUE THREAD

**Issue by interkosmos** [Fortran-FOSS-Programmers/ford#345]
Running FORD on a source file that contains regular expressions fails with the following exception:
```
$ ford  -d ./src project.md
Reading file src/dm_option.f90
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/sre_parse.py", line 1039, in parse_template
    this = chr(ESCAPES[this][1])
KeyError: '\\s'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/.local/bin/ford", line 33, in <module>
    sys.exit(load_entry_point('FORD==6.0.0', 'console_scripts', 'ford')())
  File "/home/user/.local/lib/python3.9/site-packages/ford/__init__.py", line 391, in run
    main(proj_data,proj_docs,md)
  File "/home/user/.local/lib/python3.9/site-packages/ford/__init__.py", line 342, in main
    project = ford.fortran_project.Project(proj_data)
  File "/home/user/.local/lib/python3.9/site-packages/ford/fortran_project.py", line 90, in __init__
    ford.sourceform.FortranSourceFile(os.path.join(curdir,item),settings, preprocessor, ext in self.fixed_extensions,incl_src=html_incl_src))
  File "/home/user/.local/lib/python3.9/site-packages/ford/sourceform.py", line 1052, in __init__
    FortranContainer.__init__(self,source,"")
  File "/home/user/.local/lib/python3.9/site-packages/ford/sourceform.py", line 646, in __init__
    self.modules.append(FortranModule(source,
  File "/home/user/.local/lib/python3.9/site-packages/ford/sourceform.py", line 670, in __init__
    self.subroutines.append(FortranSubroutine(source,
  File "/home/user/.local/lib/python3.9/site-packages/ford/sourceform.py", line 756, in __init__
    self.variables.extend(line_to_variables(source,line,
  File "/home/user/.local/lib/python3.9/site-packages/ford/sourceform.py", line 2272, in line_to_variables
    initial = initial[0:search_from] + QUOTES_RE.sub(string,initial[search_from:],count=1)
  File "/usr/local/lib/python3.9/re.py", line 327, in _subx
    template = _compile_repl(template, pattern)
  File "/usr/local/lib/python3.9/re.py", line 318, in _compile_repl
    return sre_parse.parse_template(repl, pattern)
  File "/usr/local/lib/python3.9/sre_parse.py", line 1042, in parse_template
    raise s.error('bad escape %s' % this, len(this))
re.error: bad escape \s at position 5
```

The error is caused by the following line:

```fortran
        character(len=*), parameter :: REGEX2 = '(?:\s+([\w-]+?)|\s*"([\w-]+?)")(?:\s|$))' 
```

Replacing `\` with `\\` in the string solves the issue, but requires additional parsing.

**Comment by ZedThree** [comment]
Hi @interkosmos, I think this was fixed in #336: your reproducer works for me on the latest `master`. We're hoping to get a new version out shortly that has this fix in!

**Comment by interkosmos** [comment]
Thank you, works with current pip version now.
